app-games/src/main/java/com/aptoide/android/aptoidegames/feature_payments/credit_card/AdyenCreditCardScreen.kt

```diff
@@ -0,0 +1,418 @@
+      AdyenCreditCardScreen(
+        packageName = "packageName",
+        onBuyClick = {},
+        onOtherPaymentMethodsClick = {},
+        onBuyClickEnabled = Random.nextBoolean(),
+        paymentView = {
+          Box(
+            modifier = Modifier
+              .padding(top = 16.dp)
+              .height(150.dp)
+              .fillMaxWidth()
+              .border(1.dp, Palette.Black),
+            contentAlignment = Alignment.Center
+          ) {
+            Text(text = "Adyen View")
+          }
+        }
+      )
+    }
+  }
+}
+
+@PreviewLandscapeLight
+@Composable
+private fun AdyenCreditCardScreenLandscapePreview() {
+  AptoideTheme {
+    AppGamesPaymentBottomSheet {
+      AdyenCreditCardScreen(
+        packageName = "packageName",
+        onBuyClick = {},
+        onOtherPaymentMethodsClick = {},
+        onBuyClickEnabled = Random.nextBoolean(),
+        paymentView = {
+          Box(
+            modifier = Modifier
+              .padding(top = 16.dp)
+              .height(150.dp)
+              .fillMaxWidth()
+              .border(1.dp, Palette.Black),
+            contentAlignment = Alignment.Center
+          ) {
+            Text(text = "Adyen View")
+          }
+        }
+      )
+    }
+  }
+}
+
+@Composable
+private fun BuildAdyenCreditCardScreen(
+  onFinish: (Boolean) -> Unit,
+  popBackStack: () -> Unit,
+  paymentMethodId: String,
+) {
+  val context = LocalContext.current as ComponentActivity
+  val lifecycleOwner = LocalLifecycleOwner.current
+  val (uiState, buy) = rememberAdyenCreditCardUIState(paymentMethodId)
+  var onBuyClick by remember { mutableStateOf<(() -> Unit)?>(null) }
+
+  var finished by remember { mutableStateOf(false) }
+
+  AdyenCreditCardStateEffect(paymentMethodId, uiState)
+
+  AppGamesPaymentBottomSheet(
+    onClick = {
+      if (uiState is Success) {
+        onFinish(true)
+        finished = true
+      }
+    },
+    onOutsideClick = {
+      onFinish(uiState is Success)
+      finished = true
+    }
+  ) {
+    when (uiState) {
+      is MakingPurchase -> LoadingView(textMessage = R.string.iap_making_purchase_title)
+
+      is Loading -> LoadingView()
+      is Error -> AdyenCreditCardErrorScreen(
+        error = uiState.error,
+        onRetryClick = popBackStack,
+        onContactUs = { SupportActivity.open(context, "payments") }
+      )
+
+      is Input -> {
+        LaunchedEffect(Unit) {
+          uiState.cardComponent(context).observe(lifecycleOwner) { cardState ->
+            onBuyClick = if (cardState.isReady && cardState.isInputValid) {
+              {
+                buy(cardState)
+              }
+            } else {
+              null
+            }
+          }
+        }
+
+        AdyenCreditCardScreen(
+          packageName = uiState.purchaseRequest.domain,
+          onBuyClickEnabled = onBuyClick != null,
+          onBuyClick = onBuyClick ?: {},
+          onOtherPaymentMethodsClick = popBackStack
+        ) { /*Adyen payment view*/
```

Comment:
If this comment is irrelevant, please remove.
If it is relevant, then what are we missing?